### PR TITLE
Bounds fuzz testing

### PR DESCRIPTION
**Pull request by dnomadb**
@sgillies I integrated hypothesis testing to check for more edge cases -- seeing the logic implemented in #101 not passing the randomized roundtrip testing with this.

### CODE REVIEW

**Comment by dnomadb** [comment]
Running the same experiment as in #101 with this branch gives:
```python
import mercantile


z = 10
tiles = [[x, y, z] for x in range(2 ** z) for y in range(2 ** z)]
btiles = [list(mercantile.bounding_tile(*mercantile.bounds(*t))) for t in tiles]

difftiles = [t for t, b in zip(tiles, btiles) if t != b]

print(len(difftiles))

0

File: mercantile/__init__.py [hunk 1]
```diff
@@ -184,7 +184,9 @@ def bounds(*tile):
     xtile, ytile, zoom = tile
     a = ul(xtile, ytile, zoom)
     b = ul(xtile + 1, ytile + 1, zoom)
-    return LngLatBbox(a[0], b[1] + LL_EPSILON, b[0] - LL_EPSILON, a[1])
+    return LngLatBbox(
+        a[0] + LL_EPSILON, b[1] + LL_EPSILON, b[0] - LL_EPSILON, a[1] - LL_EPSILON
+    )
```
Comment (sgillies) [inline mercantile/__init__.py]: @dnomadb it's only the lower right corner of the tile bounds that needs a little kick.

Comment (sgillies) [reply mercantile/__init__.py]: I'd rather not shift the upper left corner like this. Instead, we'll do the shift inside the calculation of the tile index as I'm doing in #101.

**Comment by sgillies** [comment]
@dnomadb thanks, but I'd like to close this one in favor of #101, which has a few other code improvements. Okay?

**Comment by dnomadb** [comment]
Fine with me @sgillies, intent here was to demonstrate the issue I was seeing in #101, which it accomplished!

**Pull request closed without merging**